src/linalg.jl

```diff
@@ -217,26 +217,60 @@ end
 # Norms
 _inner_eltype(v::AbstractArray) = isempty(v) ? eltype(v) : _inner_eltype(first(v))
 _inner_eltype(x::Number) = typeof(x)
-@inline _init_zero(v::StaticArray) = float(norm(zero(_inner_eltype(v))))
+@inline _init_zero(v::AbstractArray) = float(norm(zero(_inner_eltype(v))))
+
+@inline maxabs_nested(a::Number) = abs(a)
+
+function maxabs_nested(a::AbstractArray)
+    prod(size(a)) == 0 && (return _init_zero(a))
+
+    m = maxabs_nested(a[1])
+    for j = 2:prod(size(a))
+        m = max(m, maxabs_nested(a[j]))
+    end
+
+    return m
+end
+
+@generated function maxabs_nested(a::StaticArray)
```

Comment:
Do you see speedups from this generated function? Or perhaps it serves another purpose?
```
julia> for N in [3, 10, 30]
       @show N
       x = @SVector randn(N)
       @btime maximum(abs, $x)
       @btime maxabs_nested($x)  # PR, without generated path
       @btime maxabs_nested_g($x)  # with
       end
N = 3
  2.458 ns (0 allocations: 0 bytes)
  2.417 ns (0 allocations: 0 bytes)
  2.416 ns (0 allocations: 0 bytes)
N = 10
  13.652 ns (0 allocations: 0 bytes)
  13.096 ns (0 allocations: 0 bytes)
  13.652 ns (0 allocations: 0 bytes)
N = 30
  77.830 ns (0 allocations: 0 bytes)
  73.826 ns (0 allocations: 0 bytes)
  77.786 ns (0 allocations: 0 bytes)
```

Reply:
Sorry, I didn't perform performance comparison on this.  I merely thought that `@generated` version would be always better for `StaticArrays`; now that I think about it `@generated` version of `maxabs_nested()` wouldn't be needed because the function is non-allocating anyway.

If the current push passes all CI tests, I will remove the `@generated` version of `maxabs_nested()`.  Thanks!

Reply:
Might even be simpler just to write `mapreduce(abs, max, x)`. Or normInf I think. Unless you have a compelling reason to write it out for this case.

Reply:
`mapreduce(abs, max, x)` was my initial approach, but it didn't pass the unit tests for nested array `x`.  

`normInf()` sounds great.  I will use it and remove `maxabs_nested()`.

Reply:
Just pushed a change that uses `normInf()` instead of `maxabs_nested()`.  Please let me know if you have any other suggestions!